graph/graph.go

```diff
@@ -517,6 +517,9 @@ func (graph *Graph) saveSize(root string, size int64) error {
 
 // SetDigest sets the digest for the image layer to the provided value.
 func (graph *Graph) SetDigest(id string, dgst digest.Digest) error {
+	graph.imageMutex.Lock(id)
```

Comment:
... maybe, but it might not matter. And not having it in getdigest could avoid a deadlock.